light-client/src/types.rs

```diff
@@ -49,30 +49,33 @@ pub type SignedHeader = TMSignedHeader;
 pub type TrustedState = LightBlock;
 
 /// Verification status of a light block.
-#[derive(Copy, Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
 pub enum Status {
+    /// The light block has failed verification.
+    Failed,
     /// The light has not been verified yet.
     Unverified,
     /// The light block has been successfully verified.
     Verified,
     /// The light block has been successfully verified and has passed fork detection.
     Trusted,
-    /// The light block has failed verification.
-    Failed,
 }
 
 impl Status {
     /// Return a slice of all the possible values for this enum.
-    pub fn iter() -> &'static [Status] {
-        static ALL: &[Status] = &[
-            Status::Unverified,
-            Status::Verified,
-            Status::Trusted,
-            Status::Failed,
-        ];
-
+    pub fn iter() -> &'static [Self] {
+        use Status::*;
+        static ALL: &[Status] = &[Unverified, Verified, Trusted, Failed];
         ALL
     }
+
+    pub fn most_trusted(a: Self, b: Self) -> Self {
+        std::cmp::max(a, b)
+    }
+
+    pub fn least_trusted(a: Self, b: Self) -> Self {
```

Comment:
Do we foresee an application for this method?

Reply:
No, will remove it!